core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -251,6 +257,22 @@ public Expression rewriteFunctionCall(FunctionCall node, Void context, Expressio
                 }
             }
 
+            if (functionName.equals("date_trunc")
```

Comment:
It's better to add this in a dedicated rule than keep piling on this class and turning it into a kitchen sink of function translations.

Reply:
done